src/Stratis.Bitcoin.Features.MemoryPool/Rules/CheckAllInputsMempoolRule.cs

```diff
@@ -116,7 +116,8 @@ private bool CheckInputs(MempoolValidationContext context, ScriptVerify scriptVe
                         {
                             this.logger.LogTrace("(-)[FAIL_NON_MANDATORY_SCRIPT_VERIFY]");
                             // TODO: Check what this actually means in Core's logic. If it is on testnet/regtest and RequireStandard is false, is the transaction still rejected?
-                            context.State.Fail(MempoolErrors.NonMandatoryScriptVerifyFlagFailed, ctx.Error.ToString()).Throw();
+                            context.State.Fail(MempoolErrors.NonMandatoryScriptVerifyFlagFailed, 
+                                $"{ctx.Error.ToString()} : Transaction = { tx.ToHex() } : scriptVerify = { scriptVerify } : ctx.scriptVerify = { ctx.ScriptVerify }").Throw();
```

Comment:
I think it may be better to log the extra info with the logger (perhaps Debug level); there are mempool tests that directly compare against the error message text. And I believe Bitcoin Core returns specific string error codes for various mempool failures that we have tried to emulate.

Reply:
Yes move, move the hex and script flags to a log?